apps/management-app/src/components/Authenticated/CDO/WardsDetail/index.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { yupResolver } from '@hookform/resolvers/yup';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import * as yup from 'yup';
+import ControlledSelect from '@/components/Common/ControlledSelect';
+import ControlledTextField from '@/components/Common/ControlledTextField';
+import {
+  useDeleteWardsMutation,
+  useGetDistrictsQuery,
+  useLazyGetWardByIdQuery,
+  useUpdateWardMutation,
+} from '@/store/api/generalManagementApiSlice';
+import { isApiErrorResponse } from '@/store/api/helper';
+import { showError, showSuccess } from '@/utils/toast';
+import FormInputSkeleton from '../FormInputSkeleton';
+import StaticActionBar from '../StaticActionBar';
+
+interface FormData {
+  name: string;
+  district_id: number;
+}
+
+const WardsDetail = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const [getWard, { isError }] = useLazyGetWardByIdQuery();
+
+  useEffect(() => {
+    if (isError) navigate(-1);
+  }, [isError, navigate]);
+
+  const { data: districts, isLoading: districtsLoading } = useGetDistrictsQuery(
+    {},
+  );
+
+  const schema = useMemo(
+    () =>
+      yup.object({
+        name: yup.string().required("Ward's name is required"),
+        district_id: yup.number().required(),
+      }),
+    [],
+  );
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid, isLoading },
+    reset,
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+    mode: 'onChange',
+    defaultValues: async () => {
+      const ward = await getWard(parseInt(id!), true).unwrap();
+      return ward
+        ? { name: ward.name, district_id: ward.district_id }
+        : { name: '', district_id: 0 };
+    },
+  });
+
+  const [updateWard] = useUpdateWardMutation();
```

Comment:
should handle loading status, disable button, form input, ...

Reply:
Updated